bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3747,57 +3745,70 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
       LOG_DEBUG(CNSUS, "Consensus reached");
     }
     SCOPED_MDC("pp_msg_cid", ppMsg->getCid());
+    std::deque<IRequestsHandler::ExecutionRequest> accumulatedRequests;
     while (reqIter.getAndGoToNext(requestBody)) {
       size_t tmp = reqIdx;
       reqIdx++;
       if (!requestSet.get(tmp)) {
         continue;
       }
-
       ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
       if (req.flags() & EMPTY_CLIENT_FLAG) {
         continue;
       }
       SCOPED_MDC_CID(req.getCid());
       NodeIdType clientId = req.clientProxyId();
-
       uint32_t actualReplyLength = 0;
       uint32_t actualReplicaSpecificInfoLength = 0;
-      int status;
+      int executionStatus = 1;
+      accumulatedRequests.push_back(IRequestsHandler::ExecutionRequest{
+          clientId,
+          static_cast<uint64_t>(lastExecutedSeqNum + 1),
+          req.flags(),
+          std::string(req.requestBuf(), req.requestLength()),
+          std::string(ReplicaConfig::instance().getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader), 0),
+          actualReplyLength,
+          actualReplicaSpecificInfoLength,
+          executionStatus,
+          req.requestSeqNum()});
+    }
+    if (ReplicaConfig::instance().blockAccumulation) {
+      LOG_DEBUG(GL, "With accumulation");
       {
         TimeRecorder scoped_timer(*histograms_.executeWriteRequest);
-        status = bftRequestsHandler_.execute(
-            clientId,
-            lastExecutedSeqNum + 1,
-            req.flags(),
-            req.requestLength(),
-            req.requestBuf(),
-            ReplicaConfig::instance().getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader),
-            replyBuffer,
-            actualReplyLength,
-            actualReplicaSpecificInfoLength,
-            span);
+        bftRequestsHandler_.execute(accumulatedRequests, ppMsg->getCid(), span);
       }
-
-      ConcordAssertGT(actualReplyLength,
+    } else {
+      LOG_DEBUG(GL, "Without accumulation");
+      std::deque<IRequestsHandler::ExecutionRequest> singletonRequest;
```

Comment:
singletonRequest? It's a queue, no?

Reply:
The idea is to use the execute method that gets a list of requests but for 1 request.
so I'm using singletonRequest to hold one request each time and send it to the execute method.

Reply:
Since we don't want to use the old execute interface which gets all the variables one by one we want to use only the new execute which gets a list of ExecutionRequest.
In order to use the new method and keep the ability to execute requests one by one we just creating a new list called 'singleRequest' which holds 1 request each time from the whole list and then we send only this 1 request to be executed.